### PR TITLE
Adding a reference implementation for a MindMeld http client

### PR DESCRIPTION
**Pull request by minhtuev**
This client can be used to interactively query a `parse` endpoint

```
(mindmeld) ➜  mindmeld git:(feature/http-client) ✗ ipython

In [1]: from mindmeld.components.client import ConversationClient

In [2]: conv = ConversationClient()

In [3]: conv.say('hi')
Out[3]: ["Hi, I am your HR assistant. Ask me about an individual employee, company's employee demographics or general policy questions. You can say 'Is Mia married?' or 'Average salary of females' or 'When will I get my W2?'"]

In [4]: conv.say('is Mia married?')
Out[4]: ['Mia Brown is Married']

### CODE REVIEW

File: mindmeld/components/client.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+    Attributes:
+        history (list): The history of the conversation. Starts with the most recent message.
+        context (dict): The context of the conversation, containing user context.
+        default_params (dict): The default params to use with each turn. These \
+            defaults will be overridden by params passed for each turn.
+        params (dict): The params returned by the most recent turn.
+    """
+
+    _logger = mod_logger.getChild("ConversationClient")
```
Comment (vijay120) [inline mindmeld/components/client.py]: Not used :/

Comment (minhtuev) [reply mindmeld/components/client.py]: Added warning message logging below

File: mindmeld/components/client.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                msg = "\n".join(
+                    [
+                        json.dumps(item, indent=4, sort_keys=True)
+                        for item in directive["payload"]
+                    ]
+                )
+            elif directive_name == DirectiveNames.LISTEN:
+                msg = "Listening..."
+            elif directive_name == DirectiveNames.RESET:
+                msg = "Resetting..."
```
Comment (vijay120) [inline mindmeld/components/client.py]: Shouldnt we be resetting the convo here?

File: mindmeld/components/client.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    def _generate_suggestion_text(suggestion):
+        pieces = []
+        if "text" in suggestion:
+            pieces.append(suggestion["text"])
+        if suggestion["type"] != "text":
+            pieces.append("({})".format(suggestion["type"]))
+
+        return " ".join(pieces)
+
+    def reset(self):
```
Comment (vijay120) [inline mindmeld/components/client.py]: This function is not being called

**Pull request merged**